test/functional/users_controller_test.rb

```diff
@@ -201,6 +201,23 @@ def setup
     assert_equal User.find(user.id).bio, 'Hello, there!'
   end
 
+  test 'username should not be updated test_1' do
+    u = users(:bob)
+    user = users(:bob)
+    UserSession.create(user)
+    user.username = 'newval'
+    user.save
+    assert_equal u.username, 'newval'
+  end
+
+  test 'username should not be updated test_2' do
+    user = users(:bob)
+    UserSession.create(user)
+    assert_raises ActiveRecord::ActiveRecordError do
+      user.update_attribute(:username, 'new_user')
+    end
+  end  
```

Comment:
Add `assert_equal user.username, 'Bob'`here and if you want you can combine both tests and after that append the test in unit test folder. Thanks!